### PR TITLE
Feature middleware

### PR DESCRIPTION
**Pull request by ArmintasP**
- Refactored some of the code in `MessagesController` & 'ChatsController`
- Created `IMessageService` and `MessageService` that implements it.
- Crated a custom middleware & an interceptor for logging important information.

### CODE REVIEW

**Review by PauliusLu — APPROVED** [review]
Everything seems to work. Be careful merging. I think we have fixed this error. 
image

File: Karma/Views/Messages/Index.cshtml [hunk 1]
```diff
@@ -129,7 +129,7 @@ else
                                             else
                                             {
 
-                                                string? receiver = receiver = historyPost.ReceiverUserId;
+                                                string receiver = receiver = historyPost.ReceiverUserId;
```
[the remark this answers is not in the document]
Comment (ArmintasP) [reply Karma/Views/Messages/Index.cshtml:132]: Will fix that, didn't intend to duplicate it.

**Comment by ArmintasP** [comment]
> Everything seems to be great.
> 
> However, while using the log I managed to catch an exception. This happens if you create an empty message (through the `Messaging` page). Sorry I didn't see it last time.

That's weird, since we didn't have this bug before. I added an additional check, let me know if it works now without throwing an exception.

**Comment by ArmintasP** [comment]
> Everything seems to work. Be careful merging. I think we have fixed this error. image

Thanks for mentioning, hopefully there won't be any issues with this, since I didn't commit any changes to this file and didn't include it to this branch.

**Comment by baikauskaite** [comment]
> > Everything seems to be great.
> > However, while using the log I managed to catch an exception. This happens if you create an empty message (through the `Messaging` page). Sorry I didn't see it last time.
> 
> That's weird, since we didn't have this bug before. I added an additional check, let me know if it works now without throwing an exception.

All good now

**Pull request merged**